File: BBIS.Api/ClientApp/src/models/Reports/TestOrderReportDto.ts
import { SignatoryReportDto } from './../Signatories/SignatoryReportDto';
import { BloodScreeningTestOrderDto } from "../TestOrder/BloodScreeningTestOrderDto";
import { BloodTypingTestOrderDto } from "../TestOrder/BloodTypingTestOrderDto";
import { CoombsTestOrderDto } from "../TestOrder/CoombsTestOrderDto";
import { CrossMatchTestOrderDto } from "../TestOrder/CrossMatchTestOrderDto";

export interface TestOrderReportDto {
    InstitutionName: string;
    TestOrderNumber: string;
    TestCompleted: string;
    DateCreated: string;
    PatientName: string;
    BloodType: string;
    PatientIdNo: string;
    Gender: string;
    CivilStatus: string;
    PerformedBy: SignatoryReportDto;
    ReviewedBy: SignatoryReportDto;
    ValidatedBy: SignatoryReportDto;
    Phatologist: SignatoryReportDto;
    CoombsTestOrder: CoombsTestOrderDto | null;
    BloodScreeningTestOrder: BloodScreeningTestOrderDto | null;
    BloodTypingTestOrder: BloodTypingTestOrderDto | null;
    CrossMatchTestOrders: Array<CrossMatchTestOrderDto>;
}

File: BBIS.Application/DTOs/Reports/TestOrderReportDto.cs
using BBIS.Application.DTOs.TestOrder;

namespace BBIS.Application.DTOs.Reports
{
    public class TestOrderReportDto
    {
        public string InstitutionName { get; set; }
        public string TestOrderNumber { get; set; }
        public string TestCompleted { get; set; }
        public string DateCreated { get; set; }
        public string PatientName { get; set; }
        public string BloodType { get; set; }
        public string PatientIdNo { get; set; }
        public string Gender { get; set; }
        public string CivilStatus { get; set; }
        public string Remarks { get; set; }
        
        public SignatoryReportDto PerformedBy { get; set; }
        public SignatoryReportDto ReviewedBy { get; set; }
        public SignatoryReportDto ValidatedBy { get; set; }
        public SignatoryReportDto Phatologist { get; set; }

        public List<CrossMatchTestOrderDto> CrossMatchTestOrders { get; set; }
        public CoombsTestOrderDto CoombsTestOrder { get; set; }
        public BloodScreeningTestOrderDto BloodScreeningTestOrder { get; set; }
        public BloodTypingTestOrderDto BloodTypingTestOrder { get; set; }
    }

    public class SignatoryReportDto
    {
        public string Name { get; set; }
        public string LicenseNo { get; set; }
    }
}
